scripts/migrations/2020-10-archivematica_bags_migration.py

```diff
@@ -0,0 +1,246 @@
+
+                if not found_match:
+                    raise RuntimeError(f"Missing any files matching prefix: {prefix}")
+
+        def _get_bag_files():
+            file_locations = []
+            for prefix in prefix_patterns:
+                file_locations = file_locations + _filter_bag_files(prefix)
+
+            _check_expected_prefixes_exist(file_locations)
+
+            for location in file_locations:
+                filename = location.split("/")[-1]
+                save_path = f"{working_folder}/{filename}"
+                storage_s3_client.download_file(bucket, location, save_path)
+
+        def _compress_bag():
+            shutil.make_archive(working_folder, "gztar", working_folder)
+            shutil.rmtree(working_folder, ignore_errors=True)
+
+        def _generate_checksum(file_location):
+            sha256_hash = hashlib.sha256()
+
+            with open(file_location, "rb") as f:
+                for byte_block in iter(lambda: f.read(4096), b""):
+                    sha256_hash.update(byte_block)
+
+                return sha256_hash.hexdigest()
+
+        def _generate_updated_checksums():
+            files_in_need_of_update = ["bag-info.txt", "fetch.txt"]
+
+            return {
+                filename: _generate_checksum(f"{working_folder}/{filename}")
+                for filename in files_in_need_of_update
+            }
+
+        def _load_existing_checksums():
+            tag_manifest = {}
+
+            with open(f"{working_folder}/{tagmanifest_name}") as fp:
+                for _, line in enumerate(fp):
+                    split_manifest_line = line.split(" ")
+                    checksum = split_manifest_line[0].strip()
+                    filename = split_manifest_line[1].strip()
+
+                    assert checksum
+                    assert filename
+
+                    tag_manifest[filename] = checksum
+
+            return tag_manifest
+
+        def _write_new_tagmanifest():
+            existing_checksums = _load_existing_checksums()
+            new_checksums = _generate_updated_checksums()
+
+            merged_checksums = dict(existing_checksums.items() | new_checksums.items())
+
+            with open(f"{working_folder}/{tagmanifest_name}", "w") as fp:
+                for checksum, filename in merged_checksums.items():
+                    fp.write(f"{filename} {checksum}\n")
+
+        def _get_archivematica_uuid():
+            files = _filter_bag_files("data/METS.")
+
+            assert len(files) == 1
+            mets_file_with_id = files[0]
+
+            archivematica_uuid = mets_file_with_id.split("/METS.")[-1].split(".xml")[0]
+
+            assert archivematica_uuid
+            return archivematica_uuid
+
+        def _update_bag_info():
+            archivematica_uuid = _get_archivematica_uuid()
+            with open(f"{working_folder}/bag-info.txt", "a") as fp:
+                fp.write(f"Internal-Sender-Identifier: {archivematica_uuid}")
```

Comment:
Don't forget the trailing newline!

```suggestion
                fp.write(f"Internal-Sender-Identifier: {archivematica_uuid}\n")
```